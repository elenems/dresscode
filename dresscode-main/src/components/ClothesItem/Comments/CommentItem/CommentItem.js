import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

function CommentItem(props) {
  const {
    commenterId,
    commenterLastName,
    commenterName,
    commentText,
    postedAt
  } = props.comment;
  return (
    <div className="comment">
      <div className="comment-details">
        <Link to={`/users/${commenterId}`}>
          {commenterLastName} {commenterName}
        </Link>
        <span>{postedAt.slice(0, -4)}</span>
      </div>
      <p>{commentText}</p>
    </div>
  );
}

CommentItem.propTypes = {
  comment: PropTypes.object.isRequired
};

export default React.memo(CommentItem);
