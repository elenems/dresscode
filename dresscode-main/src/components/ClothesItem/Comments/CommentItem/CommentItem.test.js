import { shallow } from "enzyme";
import React from "react";
import CommentItem from "./CommentItem";

describe("CommentItem", () => {
  const props = {
    comment: {
      commenterId: "19wa28asd182s",
      commenterLastName: "Bill",
      commenterName: "Loid",
      commentText: "Give me some text",
      postedAt: "20-10-2019"
    }
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<CommentItem {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });
});
