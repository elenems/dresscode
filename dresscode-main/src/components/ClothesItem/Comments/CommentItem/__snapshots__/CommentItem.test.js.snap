// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentItem Matches snapshot 1`] = `
"<div className=\\"comment\\">
  <div className=\\"comment-details\\">
    <Link to=\\"/users/19wa28asd182s\\">
      Bill
       
      Loid
    </Link>
    <span>
      20-10-
    </span>
  </div>
  <p>
    Give me some text
  </p>
</div>"
`;
